src/middlewared/middlewared/worker.py

```diff
@@ -99,7 +101,13 @@ def main_worker(*call_args):
 
 
 def receive_events():
-    c = Client('ws+unix:///var/run/middlewared-internal.sock', py_exceptions=True)
+    while True:
+        try:
+            c = Client('ws+unix:///var/run/middlewared-internal.sock', py_exceptions=True)
+            break
+        except socket.timeout:
+            MIDDLEWARE.logger.warning("Timeout connecting to middlewared internal socket, retrying...")
```

Comment:
Why would it be unable to connect or timeout?

Reply:
Due to race condition between main process listening on socket and worker startup

Reply:
Is it hard to fix that race? Do not start process pool until after we start listening